src/main/java/xyz/nucleoid/plasmid/game/GameSpace.java

```diff
@@ -0,0 +1,72 @@
+package xyz.nucleoid.plasmid.game;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import xyz.nucleoid.plasmid.game.player.PlayerSet;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+/**
+ * Represents the space within which a game occurs through attached {@link GameLogic}
+ */
+public interface GameSpace {
+    void openGame(Consumer<GameLogic> builder);
+
+    CompletableFuture<StartResult> requestStart();
+
+    <T extends AutoCloseable> T addResource(T resource);
```

Comment:
Javadoc would be appreciated too, and why is there one for `GameSpace` and another for `GameLogic`?

Are the resources shared between those two? If not what's the difference?